### DISCUSSION TITLE
TLS configuration question

### DISCUSSION THREAD

**Discussion by zeed7000** [inspircd/inspircd#2139] [Support]
Hi,
 I installed Sectigo PositiveSSL Wildcard
I installed the following inspircd-4.4.0 + anope-2.1.8
Connecting in unsafe mode
And when trying in safe mode, errors appear?

When checking the range, this error result appears?
irc.zeed.net:7000
<html>
<body>

irc.zeed.net resolves to 77.188.46.666
--

  | Server Type: Apache

  | No SSL certificates were found on irc.chatcool.net. Make sure that the name resolves to the correct server and that the SSL port (default is 443) is open on your server's firewall.

</body>
</html>

irc.zeed.net:2083
<html>
<body>

irc.zeed.net resolves to 77.188.46.666
--

  | Server Type: Apache

  | The certificate should be trusted by all major web browsers (all the correct intermediate certificates are installed).

  | The certificate was issued by GlobalSign.  Write review of GlobalSign | The certificate was issued by GlobalSign. | Write review of GlobalSign
The certificate was issued by GlobalSign. | Write review of GlobalSign

  | The certificate will expire in 386 days.Remind me | The certificate will expire in 386 days. | Remind me
The certificate will expire in 386 days. | Remind me

  | The hostname (irc.chatcool.net) is correctly listed in the certificate.

</body>
</html>irc.zeed.net resolves to 77.188.46.666
 	
Server Type: Apache
 	
The certificate should be trusted by all major web browsers (all the correct intermediate certificates are installed).
 	
The certificate was issued by [GlobalSign](https://www.sslshopper.com/globalsign-certificate-authority-reviews.html).	  
 	
The certificate will expire in 386 days.	
 	
The hostname (irc.zeed.net) is correctly listed in the certificate.
	Common name: *.zeed.net
SANs: *.zeed.net, zeed.net
Valid from December 12, 2024 to January 13, 2026
Serial Number: 437c0ba72cc37f8eda38389d
Signature Algorithm: sha256WithRSAEncryption
Issuer: GlobalSign GCC R6 AlphaSSL CA 2023	
	 
	Common name: GlobalSign GCC R6 AlphaSSL CA 2023
Organization: GlobalSign nv-sa
Location: BE
Valid from July 18, 2023 to July 18, 2026
Serial Number: 7f1f2c902e83d0e3b6fb3bee478b5e80
Signature Algorithm: sha256WithRSAEncryption
Issuer: GlobalSign
What is the solution? Thank you.

**Comment by sadiepowell** [comment]
Whatever tool you're using is intended for HTTP not for IRC. You should use the `inspircd-testssl` tool which exists in the InspIRCd binary directory for testing InspIRCd TLS endpoints.

That said, I'm not sure what you're specifically trying to do? What is the IRC server endpoint you're trying to connect to? Is this a client or server endpoint?

**Comment by zeed7000** [comment]
Hello
Thanks for the reply
The server ircd endpoint so I can securely log in to the IRC server and the members

**Comment by zeed7000** [comment]
./bin/inspircd-testssl 77.188.46.666 6697
Name "IO::Socket::errstr" used only once: possible typo at ./bin/inspircd-testssl line 82.
Checking whether 77.188.46.666/6697 is reachable ... yes
Checking whether 77.188.46.666/6697 is using plaintext ... error

It seems like the server dropped the connection before sending anything! Make sure that:

  * The endpoint you specified is actually your IRC server.
  * If you are using a firewall incoming data on TCP port 6697 are allowed.
  * The IP address you are connecting from has not been banned from the server.
These errors appear Not available on server firewall

**Comment by zeed7000** [comment]
Using configuration file conf/anope.conf
Attempting to connect to uplink #1 77.188.46.666 (77.188.46.666/7000) with protocol InspIRCd 3+
Unable to connect to uplink #1 (77.188.46.666:7000): error:0A00010B:SSL routines::wrong version numb

**Reply by sadiepowell** [reply]
That looks like you haven't configured Anope to actually configure with TLS support. You need to set `ssl = yes` in the uplink block and load one of the SSL modules.

**Reply by zeed7000** [reply]
Hi
Thanks I did that to no avail
My mistake is that I don't know how to attach encryption files regarding inspircd 
Secure entry?
I just install it and it enters unsafe mode?

**Reply by sadiepowell** [reply]
Have you read this? https://docs.inspircd.org/tutorials/enable-tls/

Its the same config for servers.

**Reply by zeed7000** [reply]
Thank you
I will try to work on it
Delayed reply because I was busy
